### PR TITLE
Enabling unit test cases

### PR DESCRIPTION
**Pull request by shrshindeMSFT**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

Adding unit test cases for `page.navigateCrossDomain`

### Main changes in the PR:
Adding unit test cases to validate URL validation.

### Validation performed:

1. Added unit test cases.

### Unit Tests added:

> Unit tests are required for all changes. If no unit tests were added as part of this change, please explain why they aren't necessary.

Yes

### End-to-end tests added:

N/A

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

Yes

### CODE REVIEW

File: packages/teams-js/test/public/pages.spec.ts [hunk 1]
```diff
@@ -2248,6 +2280,101 @@ describe('Testing pages module', () => {
             expect(pages.navigateCrossDomain('https://valid.origin.com')).rejects.toEqual(errorNotSupportedOnPlatform);
           });
 
+          it(`pages.navigateCrossDomain should throw error with a bad origin when initialized with ${context}`, async () => {
+            await framelessPostMocks.initializeWithContext(context);
+            const promise = pages.navigateCrossDomain('https://badorigin.com');
+            const navigateCrossDomainMessage = framelessPostMocks.findMessageByFunc('navigateCrossDomain');
+            expect(navigateCrossDomainMessage).not.toBeNull();
+            framelessPostMocks.respondToMessage({
```
Comment (jekloudaMSFT) [inline packages/teams-js/test/public/pages.spec.ts:2288]: Are these testing teams-js validation logic, or are they validating that these calls throw correctly when we receive an error response?

Comment (shrshindeMSFT) [reply packages/teams-js/test/public/pages.spec.ts:2288]: Yes we are checking the URL validation logic

Comment (shrshindeMSFT) [reply packages/teams-js/test/public/pages.spec.ts:2288]: If you go and check the HUB SDK we already have it implemented

**Review by Jessica-Koch — APPROVED** [review]

File: packages/teams-js/test/public/pages.spec.ts [hunk 2]
```diff
@@ -330,6 +294,74 @@ describe('Testing pages module', () => {
             expect(pages.navigateCrossDomain('https://valid.origin.com')).rejects.toEqual(errorNotSupportedOnPlatform);
           });
 
+          it(`pages.navigateCrossDomain should throw error with a bad origin when initialized with ${context}`, async () => {
+            await utils.initializeWithContext(context);
+            const promise = pages.navigateCrossDomain('https://badorigin.com');
+            const navigateCrossDomainMessage = utils.findMessageByFunc('navigateCrossDomain');
+            validateRequestWithoutArguments(navigateCrossDomainMessage, 'navigateCrossDomain');
+
+            utils.respondToMessage(navigateCrossDomainMessage!, false);
```
Comment (erinha) [inline packages/teams-js/test/public/pages.spec.ts:303]: Just to clarify, there is no URL validation happening in the App SDK, it is all happening in the hub sdk and these unit tests are abstracting that away, right?

Comment (erinha) [reply packages/teams-js/test/public/pages.spec.ts:303]: Specifically, does the test behave differently if you pass a valid URL on line 299?

Comment (jekloudaMSFT) [reply packages/teams-js/test/public/pages.spec.ts:303]: Yes, that's the case, there's no validation in the App SDK - Shreyas and I discussed offline. But that being true, do we need so many tests validating the same behavior? (That pages.navigateCrossDomain throws the error response that it receives?)

**Review by Ella-ly — APPROVED** [review]

**Pull request merged**